pyqasm/visitor.py

```diff
@@ -759,6 +761,69 @@ def _visit_custom_gate_operation(
 
         return result
 
+    def _visit_external_gate_operation(
+        self, operation: qasm3_ast.QuantumGate, inverse: bool = False
+    ) -> list[qasm3_ast.QuantumGate]:
+        """Visit an external gate operation element.
+
+        Args:
+            operation (qasm3_ast.QuantumGate): The external gate operation to visit.
+            inverse (bool): Whether the operation is an inverse operation. Defaults to False.
+
+                            If True, the gate operation is applied in reverse order and the
+                            inverse modifier is appended to each gate call.
+                            See https://openqasm.com/language/gates.html#inverse-modifier
+                            for more clarity.
+
+        Returns:
+            list[qasm3_ast.QuantumGate]: The quantum gate that was collected.
+        """
+
+        logger.debug("Visiting custom gate operation '%s'", str(operation))
+        gate_name: str = operation.name.name
+
+        if gate_name not in self._custom_gates:
```

Comment:
I think this should be changed to include the set of basis gates which we support. Eg., if I am calling the `unroll` method like `qasm_module.unroll(external_gates = ["rx"])` and then have an `rx` gate in my program it will raise an error - 
```python 

In [27]: qasm_str = """OPENQASM 3.0;
    ...: include "stdgates.inc";
    ...: qubit[2] q;
    ...: rx(0) q[0];"""

In [28]: mod = pyqasm.load(qasm_str)

In [29]: mod.unroll(external_gates = ["rx"])
ERROR:root:Error at line 4, column 0 in QASM file
---------------------------------------------------------------------------
ValidationError                           Traceback (most recent call last)
Cell In[29], line 1
----> 1 mod.unroll(external_gates = ["rx"])

......

    783 gate_name: str = operation.name.name
    785 if gate_name not in self._custom_gates:
--> 786     raise_qasm3_error(
    787         f"Gate '{gate_name}' is not a custom gate and can not be marked as external.",
    788         span=operation.span,
    789     )
    791 gate_definition: qasm3_ast.QuantumGateDefinition = self._custom_gates[gate_name]
    792 op_qubits: list[qasm3_ast.IndexedIdentifier] = (
    793     self._get_op_bits(  # type: ignore [assignment]
    794         operation,
    795         self._global_qreg_size_map,
    796     )
    797 )

File /workspaces/pyqasm/pyqasm/exceptions.py:64, in raise_qasm3_error(message, err_type, span, raised_from)
     62 if raised_from:
     63     raise err_type(message) from raised_from
---> 64 raise err_type(message)

ValidationError: Gate 'rx' is not a custom gate and can not be marked as external.
```
The user should have the ability to specify any gate - user defined custom gates AND from the set of gates which we support, to be marked as external.

Reply:
More specifically the gates which we decompose down to simpler gates like the [`u3`](https://github.com/qBraid/pyqasm/blob/3a7fa5c0636a7b8ac5033cd9ebc100becff142df/pyqasm/maps.py#L94), [`crx`](https://github.com/qBraid/pyqasm/blob/3a7fa5c0636a7b8ac5033cd9ebc100becff142df/pyqasm/maps.py#L327), etc. will benefit from this change.